mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,509 @@
+        if let EthCoinType::Nft { .. } = self.coin_type {
+            return MmError::err(ValidatePaymentError::ProtocolNotSupported(
+                "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
+            ));
+        }
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| {
+                ValidatePaymentError::InternalError("Expected swap_v2_contracts to be Some, but found None".to_string())
+            })?;
+        validate_payment_args(args.taker_secret_hash, args.maker_secret_hash, &args.amount)
+            .map_to_mm(ValidatePaymentError::InternalError)?;
+        let maker_address = public_to_address(args.maker_pub);
+        let swap_id = self.etomic_swap_id_v2(args.time_lock, args.maker_secret_hash);
+        let maker_status = self
+            .payment_status_v2(
+                maker_swap_v2_contract,
+                Token::FixedBytes(swap_id.clone()),
+                &MAKER_SWAP_V2,
+                EthPaymentType::MakerPayments,
+                2,
+            )
+            .await?;
+
+        let tx_from_rpc = self
+            .transaction(TransactionId::Hash(args.maker_payment_tx.tx_hash()))
+            .await?;
+        let tx_from_rpc = tx_from_rpc.as_ref().ok_or_else(|| {
+            ValidatePaymentError::TxDoesNotExist(format!(
+                "Didn't find provided tx {:?} on ETH node",
+                args.maker_payment_tx.tx_hash()
+            ))
+        })?;
+        validate_from_to_and_status(
+            tx_from_rpc,
+            maker_address,
+            maker_swap_v2_contract,
+            maker_status,
+            MakerPaymentStateV2::PaymentSent as u8,
+        )?;
+
+        let validation_args = {
+            let amount = wei_from_big_decimal(&args.amount, self.decimals)?;
+            MakerValidationArgs {
+                swap_id,
+                amount,
+                taker: self.my_addr().await,
+                taker_secret_hash: args.taker_secret_hash,
+                maker_secret_hash: args.maker_secret_hash,
+                payment_time_lock: args.time_lock,
+            }
+        };
+        match self.coin_type {
+            EthCoinType::Eth => {
+                let function = MAKER_SWAP_V2.function(ETH_MAKER_PAYMENT)?;
+                let decoded = decode_contract_call(function, &tx_from_rpc.input.0)?;
+                validate_eth_maker_payment_data(&decoded, &validation_args, function, tx_from_rpc.value)?;
+            },
+            EthCoinType::Erc20 { token_addr, .. } => {
+                let function = MAKER_SWAP_V2.function(ERC20_MAKER_PAYMENT)?;
+                let decoded = decode_contract_call(function, &tx_from_rpc.input.0)?;
+                validate_erc20_maker_payment_data(&decoded, &validation_args, function, token_addr)?;
+            },
+            EthCoinType::Nft { .. } => unreachable!(),
```

Comment:
Better to return an error here instead of `unreachable!()`

Reply:
But there is already an Nft match check at the beginning of function
```rust
    pub(crate) async fn validate_taker_funding_impl(
        &self,
        args: ValidateTakerFundingArgs<'_, Self>,
    ) -> ValidateSwapV2TxResult {
        if let EthCoinType::Nft { .. } = self.coin_type {
            return MmError::err(ValidateSwapV2TxError::ProtocolNotSupported(
                "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
            ));
        }
```
even with this you still prefer to return the error one more time?

Reply:
Oke, removed panic https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/a9a5c47551c39b37fc447a9feb6971c170f0d549